pkg/cluster/internal/kubeadm/config.go

```diff
@@ -74,11 +76,14 @@ clusterName: "{{.ClusterName}}"
 bootstrapTokens:
 - token: "{{ .Token }}"
 {{ if .ControlPlaneEndpoint -}}
-controlPlaneEndpoint: {{ .ControlPlaneEndpoint }}
+controlPlaneEndpoint: "{{ .ControlPlaneEndpoint }}"
 {{- end }}
 # we use a well know port for making the API server discoverable inside docker network. 
 # from the host machine such port will be accessible via a random local port instead.
 api:
+  {{ if .APIAdvertiseAddress -}}
+  advertiseAddress: "{{ .APIAdvertiseAddress }}"
```

Comment:
why do we need to add the advertise address as part of the this PR?
`controlPlaneEndpoint` should already handle that for both single CP and multi CP scenarios.

Reply:
I had to add the `advertiseAddress ` because `etcd` , the `kube-apiserver` and other components were using the interface IPv4 address, also the `sevrer` field in `/etc/kubernetes/kubelet.conf` was using the IPv4 address

Reply:
hmm, not sure how should we approach this problem.
cc @fabriziopandini

Reply:
I have a similar issue when adding multiple control nodes with IPv6, first control node register successfully but the second one falls back to IPb4 and fails

```
EBU[00:13:45] Running: /usr/bin/docker [docker exec --privileged -t kind-control-plane2 kubeadm join [2001:db8:1::242:ac11:2]:6443 --experimental-control-plane --token abcdef.0123456789abcdef --discovery-token-unsafe-skip-ca-verification --ignore-preflight-errors=all --v=6]
⢎⡰ [control-plane2] Joining control-plane node to Kubernetes ☸
DEBU[00:15:06] I0225 23:13:46.054405    1008 join.go:299] [join] found NodeName empty; using OS hostname as NodeName
I0225 23:13:46.054436    1008 join.go:303] [join] found advertiseAddress empty; using default interface's IP address as advertiseAddress
I0225 23:13:46.054653    1008 interface.go:384] Looking for default routes with IPv4 addresses
I0225 23:13:46.054665    1008 interface.go:389] Default route transits interface "eth0"
I0225 23:13:46.054758    1008 interface.go:196] Interface eth0 is up
I0225 23:13:46.054792    1008 interface.go:244] Interface "eth0" has 3 addresses :[172.17.0.4/16 2001:db8:1::242:ac11:4/64 fe80::42:acff:fe11:4/64].
I0225 23:13:46.054808    1008 interface.go:211] Checking addr  172.17.0.4/16.
I0225 23:13:46.054814    1008 interface.go:218] IP found 172.17.0.4
I0225 23:13:46.054820    1008 interface.go:250] Found valid IPv4 address 172.17.0.4 for interface "eth0".
I0225 23:13:46.054827    1008 interface.go:395] Found active IP 172.17.0.4
[preflight] Running pre-flight checks
I0225 23:13:46.054858    1008 join.go:328] [preflight] Running general checks
```